src/main/java/org/spongepowered/api/entity/living/player/tab/TabListEntry.java

```diff
@@ -0,0 +1,189 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living.player.tab;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.profile.GameProfile;
+import org.spongepowered.api.entity.living.player.gamemode.GameMode;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents the information attached to an entry in a {@link TabList}.
+ */
+public interface TabListEntry {
+
+    /**
+     * Creates a new {@link Builder} to create {@link TabListEntry}s.
+     *
+     * @return The new builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the {@link GameProfile} associated with this entry.
+     *
+     * @return The profile associated with this entry
+     */
+    GameProfile getProfile();
+
+    /**
+     * Gets this entry's display name.
+     *
+     * @return This entry's display name
+     */
+    Optional<Text> getDisplayName();
+
+    /**
+     * Sets this entry's display name.
+     *
+     * @param displayName The new display name
+     */
+    void setDisplayName(@Nullable Text displayName);
+
+    /**
+     * Gets the latency for this entry.
+     *
+     * @return The latency for this entry
+     */
+    int getLatency();
+
+    /**
+     * Sets the latency for this entry.
+     *
+     * <p>The client displays connection bars based on this number.</p>
+     *
+     * <table summary="">
+     *     <thead>
+     *         <tr>
+     *             <th>Bars</th>
+     *             <th>Time</th>
+     *         </tr>
+     *     </thead>
+     *     <tbody>
+     *         <tr>
+     *             <td>0</td>
+     *             <td>Less than 0</td>
+     *         </tr>
+     *         <tr>
+     *             <td>1</td>
+     *             <td>1000+</td>
+     *         </tr>
+     *         <tr>
+     *             <td>2</td>
+     *             <td>600 - 999</td>
+     *         </tr>
+     *         <tr>
+     *             <td>3</td>
+     *             <td>300 - 599</td>
+     *         </tr>
+     *         <tr>
+     *             <td>4</td>
+     *             <td>150 - 299</td>
+     *         </tr>
+     *         <tr>
+     *             <td>5</td>
+     *             <td>0 - 149</td>
+     *         </tr>
+     *     </tbody>
+     * </table>
+     *
+     * @param latency The new latency, in milliseconds
+     */
+    void setLatency(int latency);
+
+    /**
+     * Gets the {@link GameMode} this entry is in.
+     *
+     * @return The gamemode this entry is in
+     */
+    GameMode getGameMode();
+
+    /**
+     * Sets this entry's gamemode. Note that setting this to spectator is what
+     * enables spectator (noclip, invisibility), this isn't just a visual change.
```

Comment:
I know you haven't touched this but this should probably a) link `GameModes.SPECTATOR` and b) the wording here sounds weird: "setting this to spectator is what enables spectator", should be probably something like "spectator effects"